### PR TITLE
ES-842

### PR DESCRIPTION
**Pull request by ase-101**

### CODE REVIEW

File: docs/design/handle_support.md [hunk 1]
```diff
@@ -30,23 +30,49 @@ As handles are revocable they provide strong privacy by default. If a user feels
...
+    - Particular fieldId is marked/unmarked as handle in the published schema, until the identity is updated using the new published schema, previously selected handles will be supported.
+
+* Change in the `mosip_idrepo.handle` table:
+    - Add `status` column in the `mosip_idrepo.handle` table.
+
+* Changes in Job: In CredentialServiceManager, when we fetch the list of handles for a UIN to issue credentials:
+    - Issue credential event to be raised if the handle status is `ACTIVATED`.
+    - If the handle status is `DELETE`, then send `REMOVE_ID` event to IDA and mark the handle status as `DELETE_REQUESTED`.
+    - If the UIN will be blocked or deactivated, then revoke credentials for all selected handles for a UIN.
+    - We should change in the IDA to acknowledge `REMOVE_ID` event to identity-service as `REMOVE_ID_STATUS` event. On receiving successful acknowledgment from `REMOVE_ID_STATUS` event, delete the entry from `mosip_idrepo.handle` table.
```
Comment (vishwa-vyom) [inline docs/design/handle_support.md:74]: should this event not be moved to a history table for audits and debugging purpose ?

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: docs/design/handle_support.md [hunk 2]
```diff
@@ -30,23 +30,49 @@ As handles are revocable they provide strong privacy by default. If a user feels
...
-	3. We should change in the IDA to acknowledge `REMOVE_ID` event to identity-service as `REMOVE_ID_STATUS` event. On receiving successful acknowledgment from `REMOVE_ID_STATUS` event, delete the entry from `mosip_idrepo.handle` table.
+
+	1. selectedHandles not present or selectedHandles is set as null in the request: ( should consider the saved selectedhandles )
+
+		a. Fields to update are marked as handle in the identity schema:
+		 - Check the saved identity object to see if any fields are selected as Handle.
+		 - If any input field is selected as handle in the saved object, update the old value as `DELETE` in the `mosip_idrepo.handle` table. And add the new value as handle by considering the previously selectedHandles list in the saved identity object.
+		 - If none of the input fields are selected as handle, proceed with (iii)
+
+		b. Fields to update are NOT marked as handle in identity schema, proceed with (iii)
```
[the remark this answers is not in the document]
Comment (ase-101) [reply docs/design/handle_support.md:41]: yeah when its rendered it is displayed properly.
image

File: docs/design/handle_support.md (hunk 1 above)
Comment (ase-101) [reply docs/design/handle_support.md:74]: We donot maintain history table for handle, we shall audit remove_id acknowledgement.

File: docs/design/handle_support.md [hunk 3]
```diff
@@ -30,23 +30,49 @@ As handles are revocable they provide strong privacy by default. If a user feels
...
-
-	In CredentialServiceManager, when we fetch the list of handles for a UIN to issue credentials:
-	1. Issue credential event to be raised if the handle status is `ACTIVATED`.
-	2. If the handle status is `DELETE`, then send `REMOVE_ID` event to IDA and mark the handle status as `DELETE_REQUESTED`.  
-	Note: If the UIN will be blocked or deactivated, then revoke credentials for all selected handles for a UIN.
-	3. We should change in the IDA to acknowledge `REMOVE_ID` event to identity-service as `REMOVE_ID_STATUS` event. On receiving successful acknowledgment from `REMOVE_ID_STATUS` event, delete the entry from `mosip_idrepo.handle` table.
+
+	1. selectedHandles not present or selectedHandles is set as null in the request: ( should consider the saved selectedhandles )
+
+		a. Fields to update are marked as handle in the identity schema:
```
[the remark this answers is not in the document]
Comment (ase-101) [reply docs/design/handle_support.md:36]: Yes we are covering this case

File: docs/design/handle_support.md (hunk 1 above)
Comment (vishwa-vyom) [reply docs/design/handle_support.md:74]: Tomorrow if there is a field compliant on one's handle used by someone else. Does the audit serve the purpose of debugging ?

File: docs/design/handle_support.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (vishwa-vyom) [reply docs/design/handle_support.md:41]: oh ok

File: docs/design/handle_support.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (vishwa-vyom) [reply docs/design/handle_support.md:36]: can you point me to the line, mentioning this behaviour in the md file?

**Pull request closed without merging**